lib/block-templates.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Block template functions.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Register a template.
+ *
+ * @param string       $template_name  Template name. It can only contain lowercase alphanumeric characters, dashes,
+ *                                     and underscores. See sanitize_key().
+ * @param array|string $args           Object type or array of object types with which the taxonomy should be associated.
+ * @param array|string $args           {
+ *     @type string        $title                 Optional. Title of the template as it will be shown in the Site Editor
+ *                                                and other UI elements.
+ *     @type string        $description           Optional. Description of the template as it will be shown in the Site
+ *                                                Editor.
+ *     @type string        $content               Optional. Default content of the template that will be used when the
+ *                                                template is rendered or edited in the editor.
+ *     @type string[]      $post_types            Optional. Array of post types to which the template should be available.
+ *     @type string        $plugin                Uri of the plugin that registers the template.
+ * }
+ * @return WP_Block_Template|WP_Error The registered template object on success, WP_Error object on failure.
+ */
+function gutenberg_register_template( $template_name, $args = array() ) {
+	return WP_Templates_Registry::get_instance()->register( $template_name, $args );
+}
+
+/**
+ * Unregister a template.
+ *
+ * @param string       $template_id  Template id in the form of `plugin_uri//template_name`.
+ * @return true|WP_Error True on success, WP_Error on failure or if the template doesn't exist.
+ */
+function gutenberg_unregister_template( $template_id ) {
+	return WP_Templates_Registry::get_instance()->unregister( $template_id );
+}
```

Comment:
Should we rename these functions to `wp_register_template` ... (I'm guessing that would be their final name in WP) and move this file into `6.7` folder?

Reply:
Also done in https://github.com/WordPress/gutenberg/pull/61577/commits/c9fdab88f92091dfc6b0eee895498e872c1c1b7b.

Reply:
> I'm guessing that would be their final name in WP

I'm not 100% sure to understand why the `wp_` prefix is needed. Is that because these functions are in the feature plugin for now? [`register_block_type()`](https://developer.wordpress.org/reference/functions/register_block_type/) doesn't have the `wp_` prefix. :thinking: